lib/puppet/parser/functions/ensure_resources.rb

```diff
@@ -0,0 +1,70 @@
+# Test whether a given class or definition is defined
+require 'puppet/parser/functions'
+
+Puppet::Parser::Functions.newfunction(:ensure_resources,
+                                      :type => :statement,
+                                      :doc => <<-'ENDOFDOC'
+Takes a resource type, title (hash/array), and a list of attributes that describe a
+resource.
+
+    user { 'dan':
+      ensure => present,
+    }
+
+This example only creates the resource if it does not already exist:
+
+    ensure_resources('user', 'dan', {'ensure' => 'present' })
+
+If the resource already exists but does not match the specified parameters,
+this function will attempt to recreate the resource leading to a duplicate
+resource definition error.
+
+An array of resources can also be passed in and each will be created with
+the type and parameters specified if it doesn't already exist.
+    ensure_resources('user', ['dan','alex'], {'ensure' => 'present'})
+
+An hash of resources can also be passed in and each will be created with
+the type and parameters specified if it doesn't already exist.
+    ensure_resources('user', {'dan' => { gid => 'mygroup', uid => '600' } ,  'alex' => { gid => 'mygroup' }}, {'ensure' => 'present'})
```

Comment:
I agree with @hunner . Since `ensure_resource()` already handles the case where you'd want to create duplicate resources with identical parameters, this new function should be restricted to support finer grain specificity to multiple resource creation.